From 16f0b179fcc36634d6c45b49f420a07798991b9c Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@chainguard.dev>
Date: Sat, 8 Nov 2025 07:27:25 +0000
Subject: [PATCH] systemd: allow getattrs on fixed devices

systemd-networkd is trying to monitor hot-plug of network devices,
which seems to getattr on the block devices.

Allow for now, but also consider dontaudit.

Need to test if hotplug of network interfaces works.
---
 policy/modules/system/systemd.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 4a4a401ca..ef3023183 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -673,6 +673,9 @@ kernel_rw_net_sysctls(systemd_networkd_t)
 kernel_read_xen_state(systemd_networkd_t)
 kernel_read_network_state(systemd_networkd_t)
 kernel_rw_psi(systemd_networkd_t)
+# Likely from https://github.com/systemd/systemd/blob/f8293452b65551c75534a35af8e6202a45837299/src/network/networkd-manager.c#L173
+# Another option is storage_dontaudit_getattr_fixed_disk_dev(systemd_networkd_t)
+storage_getattr_fixed_disk_dev(systemd_networkd_t)
 
 corenet_rw_tun_tap_dev(systemd_networkd_t)
 corenet_tcp_bind_all_nodes(systemd_networkd_t)
-- 
2.51.0

